### PR TITLE
doc(iot-dev): Add additional documentation to connection status callback

### PR DESCRIPTION
**Pull request by timtay-microsoft**
A few customers have been confused about this point, so it is worth adding some clarification around it.

### CODE REVIEW

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java [hunk 1]
```diff
@@ -420,6 +420,11 @@ public void sendReportedProperties(Set<Property> reportedProperties, Integer ver
      * with a status and a reason why the device's status changed. When the callback is fired, the provided context will
      * be provided alongside the status and reason.
      *
+     * This connection status callback is not triggered by any upstream connection change events. For example, if
```
Comment (drwill-ms) [inline device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java:423]: Are you also planning customer-facing docs?

**Review by drwill-ms — APPROVED** [review]

**Review by azabbasi — APPROVED** [review]

File: device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java (hunk 1 above)
Comment (timtay-microsoft) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java:423]: What do you mean? These are the customer facing docs. This class is "InternalClient" but the method and its javadocs are exposed via inheritance in both ModuleClient and DeviceClient

Comment (drwill-ms) [reply device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java:423]: I assumed the fact it is named internal client meant it wasn't customer facing. My mistake. :)

**Pull request merged**